src/Service/VendorService/FtpDownloadService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Service\VendorService;
+
+use App\Exception\DownloadFailedException;
+
+/**
+ * Class FtpDownloadService.
+ */
+class FtpDownloadService
+{
+    /**
+     * Download file from remote ftp server.
+     *
+     * @param string $host
+     * @param string $username
+     * @param string $password
+     * @param string $root
+     * @param string $localArchive
+     * @param string $remoteArchive
+     *
+     * @throws DownloadFailedException
+     */
+    public function download(string $host, string $username, string $password, string $root, string $localArchive, string $remoteArchive): void
```

Comment:
This could be a static function. No reason to pass `FtpDownloadService` as a service